docs/search/elasticsearch-component.md

```diff
@@ -0,0 +1,187 @@
+
+```xml
+<PackageReference Include="Aspire.Elastic.Clients.Elasticsearch"
+                  Version="[SelectVersion]" />
+```
+
+---
+
+For more information, see [dotnet add package](/dotnet/core/tools/dotnet-add-package) or [Manage package dependencies in .NET applications](/dotnet/core/tools/dependencies).
+
+## Example usage
+
+In the _Program.cs_ file of your project, call the `AddElasticsearchClient` extension method to register a `ElasticsearchClient` for use via the dependency injection container. The method takes a connection name parameter.
+
+```csharp
+builder.AddElasticsearchClient("elasticsearch");
+```
+
+## App host usage
+
+To model the RabbitMQ resource in the app host, install the [Aspire.Hosting.Elasticsearch](https://www.nuget.org/packages/Aspire.Hosting.Elasticsearch) NuGet package in the [app host](xref:aspire/app-host) project.
+
+### [.NET CLI](#tab/dotnet-cli)
+
+```dotnetcli
+dotnet add package Aspire.Hosting.Elasticsearch
+```
+
+### [PackageReference](#tab/package-reference)
+
+```xml
+<PackageReference Include="Aspire.Hosting.Elasticsearch"
+                  Version="[SelectVersion]" />
+```
+
+---
+
+In the _Program.cs_ file of `AppHost`, register a Elasticsearch cluster and consume the connection using the following methods:
+
+```csharp
+var elasticsearch = builder.AddElasticsearch("elasticsearch");
+
+var myService = builder.AddProject<Projects.MyService>()
+                       .WithReference(elasticsearch);
+```
+
+The `WithReference` method configures a connection in the `MyService` project named `elasticsearch`. In the _Program.cs_ file of `MyService`, the Elasticsearch connection can be consumed using:
+
+```csharp
+builder.AddElasticsearchClient("elasticsearch");
+```
+
+## Configuration
+
+The .NET Aspire Elasticsearch client component provides multiple options to configure the server connection based on the requirements and conventions of your project.
+
+### Use a connection string
+
+When using a connection string from the `ConnectionStrings` configuration section, you can provide the name of the connection string when calling `builder.AddElasticsearchClient()`:
+
+```csharp
+builder.AddElasticsearchClient("elasticsearch");
+```
+
+And then the connection string will be retrieved from the `ConnectionStrings` configuration section:
+
+```json
+{
+  "ConnectionStrings": {
+    "elasticsearch": "http://elastic:password@localhost:27011"
+  }
+}
+```
+
+### Use configuration providers
+
+The .NET Aspire Elasticsearch Client component supports [Microsoft.Extensions.Configuration](/dotnet/api/microsoft.extensions.configuration). It loads the  `ElasticClientsElasticsearchSettings` from configuration by using the `Aspire:Elastic:Clients:Elasticsearch` key. Example _appsettings.json_ that configures some of the options:
+
+```json
+{
+  "Aspire": {
+    "Elastic": {
+      "Clients": {
+        "Elasticsearch": {
+            "ConnectionString": "http://elastic:password@localhost:27011"
```

Comment:
```suggestion
            "Endpoint": "http://elastic:password@localhost:27011"
```

Reply:
Sorry, saw the comment with link to aspire updates before I saw your suggestions. I fixed it.